### ISSUE TITLE
There is a gap between the picture and the picture

### ISSUE THREAD

**Issue by Bedroome** [flutter/flutter#65368]
```dart
class A extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return ListView.builder(
      itemBuilder: (context, index) {
        return Container(
          color: Colors.red,
          child: Image.network(
            'https://picsum.photos/seed/${Random().nextInt(999)}/750/435',
          ),
        );
      },
      itemCount: 20,
    );
  }
}
```
<img width="483" alt="截屏2020-09-08 下午3 31 05" src="https://user-images.githubusercontent.com/35730858/92446676-be8f7200-f1e8-11ea-8e42-9b5f3db9592a.png">

```
[✓] Flutter (Channel stable, 1.20.2, on Mac OS X 10.15.6 19G2021, locale zh-Hans-CN)
 
[!] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    ! Some Android licenses not accepted.  To resolve this, run: flutter doctor --android-licenses
[✓] Xcode - develop for iOS and macOS (Xcode 11.7)
[✓] Android Studio (version 3.5)
[!] IntelliJ IDEA Ultimate Edition (version 2019.3.4)
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
[✓] VS Code (version 1.48.1)
[✓] Connected device (1 available)
```

In my example, I found that there is a gap between the picture and the picture. The sliding will appear from time to time. A red line is marked in the green circle of the picture (set to red for clarity), which is obvious Can show this problem, is there any way to solve it

**Comment by pedromassangocode** [comment]
Hi @Bedroome 
Can you please provide a minimal complete reproducible code sample? I need a code that I can just copy/paste in a `main.dart` file and run it to reproduce the issue
Thank you

**Comment by Bedroome** [comment]
> @Bedroome，您好：您
> 能提供一个最小的完整可复制代码示例吗？我需要一个可以复制/粘贴到`main.dart`文件中并运行它以重现该问题的代码，
> 谢谢
At the beginning, class A is already provided

**Comment by Bedroome** [comment]
I tried when the size is even, there will be no gaps, and there will be singular

**Comment by Bedroome** [comment]
When I try the size is even, there will be no gaps, when the size is singular, there will be

**Comment by pedromassangocode** [comment]
I was able to reproduce this on latest `master` channel. The background color is being visible between the images.

I found some (possible) related issues:
https://github.com/flutter/flutter/issues/44304
https://github.com/flutter/flutter/issues/44309
https://github.com/flutter/flutter/issues/40947
https://github.com/flutter/flutter/issues/47529

Solution at https://github.com/flutter/flutter/issues/19648#issuecomment-406868390 didn't works around this one.

![Screen Shot 2020-09-08 at 10 55 38](https://user-images.githubusercontent.com/67424416/92462088-eb7c5e80-f1c1-11ea-929c-28bfa3c19a1e.png)

<details><summary>sample code</summary>

```
import 'dart:math';
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      home: Scaffold(
        body: ListView.builder(
          itemBuilder: (context, index) {
            return Container(
              color: Colors.red,
              child: Image.network(
                'https://picsum.photos/seed/${Random().nextInt(999)}/750/435',
              ),
            );
          },
          itemCount: 20,
        ),
      ),
    );
  }
}
```
</details>

<details><summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel master, 1.22.0-10.0.pre.88, on Mac OS X 10.15.6 19G2021, locale en)
    • Flutter version 1.22.0-10.0.pre.88 at /Users/pedromassango/dev/SDKs/flutter_master
    • Framework revision e10bdbbd9d (15 hours ago), 2020-09-07 11:40:03 -0700
    • Engine revision ac8b9c4c52
    • Dart version 2.10.0 (build 2.10.0-86.0.dev)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.1)
    • Android SDK at /Users/pedromassango/Library/Android/sdk
    • Platform android-30, build-tools 30.0.1
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.7)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.7, Build version 11E801a
    • CocoaPods version 1.9.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] IntelliJ IDEA Community Edition (version 2020.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 48.1.4
    • Dart plugin version 202.7206

[✓] VS Code (version 1.48.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.13.2

[✓] Connected device (4 available)
    • sdk gphone x86 arm (mobile) • emulator-5554 • android-x86    • Android 11 (API 30) (emulator)
    • macOS (desktop)             • macos         • darwin-x64     • Mac OS X 10.15.6 19G2021
    • Web Server (web)            • web-server    • web-javascript • Flutter Tools
    • Chrome (web)                • chrome        • web-javascript • Google Chrome 85.0.4183.83

! Doctor found issues in 1 category.
```
</details>

**Comment by Bedroome** [comment]
I found that it only appears when the size is odd, and the picture does not seem to be filled with the odd size, thus showing this gap

**Comment by pedromassangocode** [comment]
Hi @Bedroome 
You can try the solution at https://github.com/flutter/flutter/issues/44304#issuecomment-550545902 and see if it works for you.

**Comment by HansMuller** [comment]
This may be another example of the known problem with antialiasing and mapping from logical pixels to device pixels. There's an open issue for a documentation fix: #14288

More info about the problem can be found here:
https://github.com/flutter/flutter/issues/14288#issuecomment-361375791
https://github.com/flutter/flutter/issues/17084#issuecomment-385718108
https://github.com/flutter/flutter/issues/15035#issuecomment-370028740

**Comment by adam-fu** [comment]
Setting Image property to `isAntiAlias: true,` made the gap barely visible. Maybe it will work for you as well.

**Comment by escamoteur** [comment]
@HansMuller I'm not sure if this is only an antialiasing problem. I encountered the same problem and it seems it's much stronger if its images than if you use coloured containers. I experienced it with aligning map tiles by using a Stack and Positioned and was only able to solve it by letting the tiles overlapp a little bit.
I tried to replicate this by using coloured containers in a different project but was not able to do do.

**Comment by caduandrade** [comment]
Even or odd, makes no difference in layouts (#78475)

**Comment by Tonku** [comment]
This issue must be fixed with flutter 3.3

**Comment by maheshmnj** [comment]
Hi @Bedroome,  I verified the issue on stable 3.3 and master 3.7 and I cannot reproduce it. I believe the issue is fixed.

![simulator_screenshot_AC3B9586-79C8-44B9-AA36-C21613299E9A](https://user-images.githubusercontent.com/31410839/210025661-e5c69f27-2051-4aa7-8b75-44d4a54d5f58.png)

<details>
<summary>code samplev</summary>

```dart
import 'dart:math';

import 'package:flutter/material.dart';

void main() => runApp(const MyApp());

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      home: Scaffold(
        body: ListView.builder(
          itemBuilder: (context, index) {
            return Container(
              color: Colors.red,
              child: Image.network(
                // random avatar size 100x100

                'https://picsum.photos/seed/${Random().nextInt(999)}/750/435',
              ),
            );
          },
          itemCount: 20,
        ),
      ),
    );
  }
}

```
</details> 


<details>
<summary>flutter doctor -v (mac)</summary>

```
[✓] Flutter (Channel master, 3.7.0-3.0.pre.33, on macOS 12.6 21G115 darwin-arm64, locale en-IN)
    • Flutter version 3.7.0-3.0.pre.33 on channel master 
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 5201856805 (38 minutes ago), 2022-12-05 18:27:21 -0800
    • Engine revision a309d239c4
    • Dart version 2.19.0 (build 2.19.0-463.0.dev)
    • DevTools version 2.20.0
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly
      to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0-rc4)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-33, build-tools 33.0.0-rc4
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14A400
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 61.2.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.70.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.55.20221129

[✓] Connected device (3 available)
    • iPhone 12 Pro (mobile) • 026D5789-9E78-4AD5-B1B2-3F8D4E7F65E4 • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)
    • macOS (desktop)        • macos                                • darwin-arm64   • macOS 12.6 21G115 darwin-arm64
    • Chrome (web)           • chrome                               • web-javascript • Google Chrome 108.0.5359.94

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```
```
[✓] Flutter (Channel stable, 3.3.9, on macOS 12.6 21G115 darwin-arm, locale en-IN)
    • Flutter version 3.3.9 on channel stable at /Users/mahesh/Development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision b8f7f1f986 (24 hours ago), 2022-11-23 06:43:51 +0900
    • Engine revision 8f2221fbef
    • Dart version 2.18.5
    • DevTools version 2.15.0

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0-rc4)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-33, build-tools 33.0.0-rc4
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14A400
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 61.2.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.70.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.53.20221101

[✓] Connected device (3 available)
    • iPhone 12 Pro (mobile) • 026D5789-9E78-4AD5-B1B2-3F8D4E7F65E4 • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)
    • macOS (desktop)        • macos                                • darwin-arm64   • macOS 12.6 21G115 darwin-arm
    • Chrome (web)           • chrome                               • web-javascript • Google Chrome 107.0.5304.110

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```
</details>

Closing as fixed. Incase anyone still encounters the issue please feel free to write in the comments and we will reopen it.

Thank you.
